src/main/java/VladMaltsev/weatherapp/controller/WeatherRESTController.java

```diff
@@ -35,154 +40,28 @@ public WeatherRESTController(WeatherDaySnapshotService weatherDaySnapshotService
     @GetMapping(value = "/getsingleday")
     public byte[] getSingleDay(@RequestParam(name = "city") String city,
                                @RequestParam(name = "country") String country,
-                               @RequestParam(name = "date") LocalDate date, Model model) throws IOException {
-
-        String data = getDataFromPage(city, country, date);
-
-        List<WeatherDaySnapshot> weatherDaySnapshot = parseData(data);
-        weatherDaySnapshotService.insertNewWeatherSnapshot(weatherDaySnapshot);
-
-        List<WeatherDuringDay> w = parseDataDuringDay(data, weatherDaySnapshot, 0);
+                               @RequestParam(name = "date") LocalDate date) throws IOException {
+        log.debug("/getsingleday getData(): "+city+" "+country+" "+date);
+        String data = getData(city, country, date);
+        List<WeatherDaySnapshotDTO> weatherDaySnapshotDTO = parseDataForTwoWeeks(data);
+        weatherDaySnapshotDTO = weatherDaySnapshotService.insertNewWeatherSnapshot(weatherDaySnapshotDTO);
+        log.debug("WeatherDaySnapshotDTO in Controller " + weatherDaySnapshotDTO.get(0));
+        List<WeatherDuringDayDTO> w = parseDataDuringDay(data, weatherDaySnapshotDTO, 0);
```

Comment:
make `DataParser` methods instance, not static. 

Add unit tests for parser and everything else.